src/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -2177,6 +2178,59 @@ public void GetObjectData_NullInfo_ThrowsArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException>("info", () => ((ISerializable)DateTime.Now).GetObjectData(null, new StreamingContext()));
         }
 
+        [Fact]
+        public void TestRoundTrippingDateTimeAndFileTime()
+        {
+            // This test ensure the round tripping of DateTime with the system file time.
+            // It is important to have this working on systems support leap seconds as the conversion wouldn't be simple
+            // conversion but involve some OS calls to ensure the right conversion is happening.
+
+            DateTime now = DateTime.UtcNow;
+            long fileTime = now.ToFileTimeUtc();
+            DateTime roundTrippedDateTime = DateTime.FromFileTimeUtc(fileTime);
+            Assert.Equal(now, roundTrippedDateTime);
+
+            now = DateTime.Now;
+            fileTime = now.ToFileTime();
+            roundTrippedDateTime = DateTime.FromFileTime(fileTime);
+            Assert.Equal(now, roundTrippedDateTime);
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void TestTimeSynchronizationWithTheSystem()
+        {
+            // The reported time by the framework should be synchronized with the OS.
+            // There shouldn't be any shift by more than one second, otherwise there is something wrong.
+            // This test is useful when running on a system supporting leap seconds to ensure when the system
+            // has leap seconds, the framework reported time will still be synchronized.
+
+            SYSTEMTIME st;
+            GetSystemTime(out st);
+            DateTime dt = DateTime.UtcNow;
```

Comment:
The test assumes these two calls will occur within a second from each other.  Any concern about that being flaky in CI on a heavily loaded system?  Worth a retry?

Reply:
I think it is worth to try.

Reply:
If we see the test flaky in CI, we can do some workaround at that time. let's try it first and look what we'll get.

Reply:
Similar patterns have *always* proved to be flaky in the CI. I am pretty sure this is going to be flaky as written.

Reply:
Ok, I'll try to make it more reliable then.

Reply:
CI timing is variable  enough and so many tests run every day that tests need to be as close to bullet proof as possible.